analysis/metric_space.lean

```diff
@@ -386,29 +386,25 @@ have tendsto (λp:α×α, dist p.1 p.2) (nhds (a, b)) (nhds (dist a b)),
   from continuous_iff_tendsto.mp continuous_dist' (a, b),
 (hf.prod_mk hg).comp (by rw [nhds_prod_eq] at this; exact this)
 
-lemma tendsto_iff_dist_tendsto_zero { X Y : Type*} [metric_space X] [metric_space Y]
-{f : X → Y} {x : X} {y : Y}: (tendsto f (nhds x) (nhds y)) ↔ (tendsto (λ x', dist (f x') y) (nhds x) (nhds 0)) :=
+@[simp]
+lemma dist_dist_0 [metric_space β] {f : α → β} {x : α} {y : β}  :
+  dist (dist (f x) y) 0 = dist (f x) y :=
+calc dist (dist (f x) y) 0 = abs ((dist (f x) y) - 0) : rfl
+  ... = abs (dist (f x) y) : by simp
+  ... = dist (f x) y : abs_of_nonneg dist_nonneg
+
+lemma tendsto_iff_dist_tendsto_zero [metric_space β] {f : α → β} {x : α} {y : β} :
+  (tendsto f (nhds x) (nhds y)) ↔ (tendsto (λ x', dist (f x') y) (nhds x) (nhds 0)) :=
```

Comment:
Another point I just saw: `nhds x` can be generalized:
* remove `[metric_space β]` (swap `β` and `α` as `α` is usually the metric space in this theory),
* instead of `nhds x` add the parameter `l : filter β` (`l` for limit) and `a : α`,
* proof `(tendsto f l (nhds y)) ↔ (tendsto (λ x', dist (f x') y) l (nhds 0))`.
After all we want to have this rule also for sequences etc. In many basic analysis rules the "input" filter `l` can be arbitrary. And this is one of the central points of using filters.

Reply:
I remember the reason now. I agree the statement should be the one you wrote. But my proof relies on Mario's `tendsto_nhds_of_metric`. I would need a version with a filter on the source for that theorem as well. I'll try to work out this, but feel free to help here. 

I can already say that `tendsto_dist` assumes a useless topological structure in https://github.com/leanprover/mathlib/blob/ff4af0d4bd7942d2d14083a1a787af26e66d91db/analysis/metric_space.lean#L382

Reply:
Ah I see what's going on. I will take care of it.

And don't worry about finding time to answer to comments, I also only find time each couple of days to work on it.

Reply:
Thank you very much!